File: Backend/models/Course.js
// models/Course.js
const Sequelize = require('sequelize');
const sequelize = require('../config/database');

const Course = sequelize.define('Courses', {
  course_id: {
    type: Sequelize.STRING(10),
    primaryKey: true,
  },
  name: {
    type: Sequelize.STRING(100),
    allowNull: false,
  },
  credits: {
    type: Sequelize.INTEGER,
    allowNull: false,
  },
  category: {
    type: Sequelize.STRING(50),
    allowNull: false,
  },
}, {
  timestamps: false,
  tableName: 'Courses',
});

module.exports = Course;

File: grade-simulator-service/dummy_data/models/course.py
from sqlalchemy import Column, String, Integer
from db import Base

class Course(Base):
    __tablename__ = "Courses"
    
    course_id = Column(String(10), primary_key=True)
    name = Column(String(100), nullable=False)
    credits = Column(Integer, nullable=False)
    category = Column(String(50), nullable=False)
    
    def to_dict(self):
        return {
            "course_id": self.course_id,
            "name": self.name,
            "credits": self.credits,
            "category": self.category
        } 